src/util-decode-mime.c

```diff
@@ -1,3580 +0,0 @@
- *
- * \param entity The entity
- *
- * \return URL entry or NULL if the operation fails
- *
- */
-static MimeDecUrl * MimeDecAddUrl(MimeDecEntity *entity, uint8_t *url, uint32_t url_len, uint8_t flags)
-{
-    MimeDecUrl *node = SCMalloc(sizeof(MimeDecUrl));
-    if (unlikely(node == NULL)) {
-        return NULL;
-    }
-    memset(node, 0x00, sizeof(MimeDecUrl));
-
-    node->url = url;
-    node->url_len = url_len;
-    node->url_flags = flags;
-
-    /* If list is empty, then set as head of list */
-    if (entity->url_list == NULL) {
-        entity->url_list = node;
-    } else {
-        /* Otherwise add to beginning of list since these are out-of-order in
-         * the message */
-        node->next = entity->url_list;
-        entity->url_list = node;
-    }
-
-    return node;
-}
-
-/**
- * \brief Creates and adds a child entity to the specified parent entity
- *
- * \param parent The parent entity
- *
- * \return The child entity, or NULL if the operation fails
- *
- */
-MimeDecEntity * MimeDecAddEntity(MimeDecEntity *parent)
-{
-    MimeDecEntity *curr, *node = SCMalloc(sizeof(MimeDecEntity));
-    if (unlikely(node == NULL)) {
-        return NULL;
-    }
-    memset(node, 0x00, sizeof(MimeDecEntity));
-
-    /* If parent is NULL then just return the new pointer */
-    if (parent != NULL) {
-        if (parent->child == NULL) {
-            parent->child = node;
-        } else {
-            curr = parent->child;
-            while (curr->next != NULL) {
-                curr = curr->next;
-            }
-            curr->next = node;
-        }
-    }
-
-    return node;
-}
-
-/**
- * \brief Creates a mime header field and fills in its values and adds it to the
- * specified entity
- *
- * \param entity Entity in which to add the field
- * \param name String containing the name
- * \param nlen Length of the name
- * \param value String containing the value
- * \param vlen Length of the value
- *
- * \return The field or NULL if the operation fails
- */
-static MimeDecField * MimeDecFillField(MimeDecEntity *entity, uint8_t *name,
-        uint32_t nlen, const uint8_t *value, uint32_t vlen)
-{
-    if (nlen == 0 && vlen == 0)
-        return NULL;
-
-    MimeDecField *field = MimeDecAddField(entity);
-    if (unlikely(field == NULL)) {
-        return NULL;
-    }
-
-    if (nlen > 0) {
-        /* convert to lowercase and store */
-        uint32_t u;
-        for (u = 0; u < nlen; u++)
-            name[u] = u8_tolower(name[u]);
-
-        field->name = (uint8_t *)name;
-        field->name_len = nlen;
-    }
-
-    if (vlen > 0) {
-        field->value = (uint8_t *)value;
-        field->value_len = vlen;
-    }
-
-    return field;
-}
-
-/**
- * \brief Pushes a node onto a stack and returns the new node.
- *
- * \param stack The top of the stack
- *
- * \return pointer to a new node, otherwise NULL if it fails
- */
-static MimeDecStackNode * PushStack(MimeDecStack *stack)
-{
```

Comment:
What feature do we need for mime inside mime ?

Reply:
I don't know, same as with C?

Reply:
As far as I get the C version, there seems to be complex stuff that are done and then unused...